src/web/src/components/Footer.vue

```diff
@@ -46,21 +49,42 @@
 
 import { getSemesters } from '@/services/YacsService';
 
+import { getSemester }  from '@/services/AdminService';
+
 export default {
     name: 'Footer',
     data() {
       return {
-        semesterOptions: []
+        semesterOptions: [],
+        currentSemester: ''
       }
     },
     methods: {
     },
     created () {
-    getSemesters().then(({ data }) => {
-            this.semesterOptions.push(...data.map(s => ({text: s.semester, value: s.semester})));
-            });
+      if(this.$route.query.semester){
+        this.currentSemester = this.$route.query.semester;
+      }
+      else{
+        getSemester().then(semester => {
+          this.currentSemester = semester[0].semester;
```

Comment:
Please change return from backend, having to chose first record of options then .semester is strange.
getDefualtSemeste should return default semester not rows of objects, i.e. "FALL 2020"  instead of [{"semester": "FALL 2020"}]

Reply:
done